lib/store/redis.go

```diff
@@ -0,0 +1,70 @@
+package store
+
+import (
+	"strings"
+	"time"
+
+	"github.com/go-redis/redis"
+)
+
+// RedisStore is a storage engine that writes to redis
+type RedisStore struct {
+	client redis.Client
+}
+
+// NewRedisClient creates a new redis client object
+func NewRedisClient(addr string) redis.Client {
+	client := redis.NewClient(&redis.Options{
+		Addr:     addr,
+		Password: "", // no password set
+		DB:       0,  // use default DB
+	})
+
+	_, err := client.Ping().Result()
+	// FIXME
+	if err != nil {
+		panic(err)
+	}
+	return *client
+}
+
+// NewRedisStore creates new store
+func NewRedisStore(client redis.Client) RedisStore {
+	return RedisStore{
+		client: client,
+	}
+}
+
+// WriteUser will write a user object to redis
+func (s RedisStore) WriteUser(user User) {
+	data := make(map[string]interface{})
+	data["username"] = user.Username
+	data["access"] = user.AccessToken
+	data["refresh"] = user.RefreshToken
+	data["updated"] = user.Updated.Format("01-02-2006")
+	s.client.HMSet("user:"+user.ID, data)
+}
+
+// GetUser will load a user from redis
+func (s RedisStore) GetUser(id string) *User {
+	data, err := s.client.HGetAll("user:" + id).Result()
+	// FIXME - return err
+	if err != nil {
+		panic(err)
```

Comment:
the convention is returning nil, err all the way back to the top so error handling can be done in a single placement.

But I also wanted to keep to your conventions.

Reply:
My convention is bad and I should feel bad. I respect you not wanting to go through that refactor though.

Reply:
Oh I will eventually. Like I said earlier I want to use this as an excuse to learn more go. Learn testing practices. Etc

I already learned a ton about interfaces.

This redis change lets me run the system locally in my k8s but not needing a filesystems.